src/CommunityCommons/javasource/communitycommons/ORM.java

```diff
@@ -176,10 +176,8 @@ private static void duplicateReverseAssociations(IContext ctx, IMendixObject src
 					throw new IllegalArgumentException("It is not possible to clone reverse referencesets: '" + fullAssocName + "'");
 				}
 				
-				List<IMendixObject> objs = Core.createXPathQuery("//$relationParent[$assocname=$value]")
-					.setVariable("relationParent", relationParent.getName())
-					.setVariable("assocname", assocname)
-					.setVariable("value", String.valueOf(src.getId().toLong()))
+				List<IMendixObject> objs = Core.createXPathQuery(String.format("//%s[%s=$value]", relationParent.getName(), assocname))
+					.setVariable("value", src)
```

Comment:
Why did you replace `src.getId().toLong()` by `src`? Will `setVariable()` automatically replace `src` by the proper id?

Reply:
Short answer: Yes it perform `.getId` or return null if MendixObject is null.

The underlying code is working with IMendixIdentifier, which is better in my opinion rather than working with String.
`setVariable` itself has several overloads and one of them is accepting `IMendixObject`. The more information we give to the xpathQuery, the better it is.
In retrieve action in microflow, you would do `[Person_Address=$newPerson]` for xpath constraint, so above code is align with this construction.